From 12bedd94508f8764430e6bec158bbca815d6ff07 Mon Sep 17 00:00:00 2001
From: Lukas Czerner <lczerner@redhat.com>
Date: Thu, 23 Oct 2014 09:53:47 -0400
Subject: [fs] ext4: change ext4_read_inline_dir() to return 0 on success

Message-id: <1414058125-4183-74-git-send-email-lczerner@redhat.com>
Patchwork-id: 98154
O-Subject: [RHEL 7.1 PATCH v2 073/171] ext4: change ext4_read_inline_dir() to return 0 on success
Bugzilla: 1150139
RH-Acked-by: Eric Sandeen <sandeen@redhat.com>

From: BoxiLiu <lewis.liulei@huawei.com>

BZ 1150139
https://bugzilla.redhat.com/show_bug.cgi?id=1150139

BREW 8142710
https://brewweb.devel.redhat.com/taskinfo?taskID=8142710

Upstream commit 48ffdab1c1eb87fee9ed7366472e2b9aa4577a60

 In ext4_read_inline_dir(), if there is inline data, the successful
 return value is the return value of ext4_read_inline_data().  Howewer,
 this is used by ext4_readdir(), and while it seems harmless to return
 a positive value on success, it's inconsistent, since historically
 we've always return 0 on success.

 Signed-off-by: BoxiLiu <lewis.liulei@huawei.com>
 Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>
 Acked-by: Tao Ma <boyu.mt@taobao.com>
 Signed-off-by: Lukas Czerner <lczerner@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/ext4/inline.c b/fs/ext4/inline.c
index b8a0746..e78e362 100644
--- a/fs/ext4/inline.c
+++ b/fs/ext4/inline.c
@@ -1443,6 +1443,7 @@ int ext4_read_inline_dir(struct file *filp,
  if (ret < 0)
   goto out;
 
+ ret = 0;
  sb = inode->i_sb;
  stored = 0;
  parent_ino = le32_to_cpu(((struct ext4_dir_entry_2 *)dir_buf)->inode);
-- 
1.7.1